### PR TITLE
fix(rate-limit): Codex の数字に持ち主を出し、API 応答前の statusLine を答えと解釈しない (#1161)

### PR DESCRIPTION
**Pull request by isamu**
## Summary

ユーザー報告「Codex の usage が反映されない／Claude の 5 時間が取れていない」の原因を調べ、2 つのバグを直した。

1. **ヘッダーの `7d 71%` は Claude ではなく Codex のものだった** — `claude usage n/a` の注記は Claude の窓が 1 本も無いときしか出ないので、隣の数字は Codex のものしかあり得ない。しかしエージェント記号は「両エージェントが数字を持つとき」しか描かれず、注記が出ている場面では必ず無印になるため、Codex の数字が Claude のものと読めた。注記と記号を 1 つの `rateLimitReadout()` で決めるようにした。
2. **プローブ最初の statusLine（API 応答前・窓なし）を「API キー課金」と断定していた** — 断定した結果、プローブが完走しなかった場合に誤った理由が出続け、失敗として数えられず再試行が 1 時間間隔に固定されていた。`cost.total_api_duration_ms` で両者を判別するようにした。

なお現行の Codex はそもそも 5h ウィンドウを出さない（実 rollout 全件で `primary.window_minutes: 10080` / `secondary: null`）。上流仕様であり、こちらは変更していない。

Refs #1161

## Items to Confirm / Review

- **`cost.total_api_duration_ms` を「API 応答があったか」の判別材料にしてよいか。** 実 PTY で Claude Code 2.1.220 のプローブを再現し、statusLine の発火ごとにペイロードを保存して確認した:

  | 発火 | `total_api_duration_ms` | `rate_limits` |
  | --- | --- | --- |
  | 1 回目（API 応答前） | `0` | ABSENT |
  | 2 回目（API 応答後） | `2769` | `five_hour` / `seven_day` あり |

  これは上流の内部フィールドなので、将来消える／意味が変わる可能性はある。読めない・欠けている場合は `false`（＝ 窓の不在から何も結論しない）に倒してある。倒す向きの根拠: 誤って `false` にした場合のコストは「再試行が遅くなり、理由の文言が曖昧になる」だけだが、誤って `true` にすると今回のバグそのものが再発する。
- **`report(agent, …)` を `reportCodex` / `reportClaudeStatus` に分けた。** Claude 側の判定を迂回できる引数を無くすためだが、公開 API の形が変わっている。
- **報告者の Claude 側がどの probe state かは未確定。** スクリーンショットからは `no-claude` / `no-windows` / `no-report` のどれかまでしか絞れない（ツールチップが写っていない）。2 の修正は `no-windows` に誤って貼り付く経路を塞ぐもので、報告者の症状がその経路だったと確認できたわけではない。1 の誤読は確定。
- `createRateLimitStore` が lint の行数上限に当たったため、状態遷移を `claudeStatusVerdict` / `afterSilence` / `afterAvailability` / `answeredDuringAttempt` に切り出し、`probeState` / `isProbing` を arrow プロパティにした。挙動は変えていない。

## User Prompt

> `Screenshot 2026-07-31 at 7.00.06.png`
>
> ユーザからの報告で、ClaudeとCodexを動かしているのですが、Codexの方はusage値が反映されず、Claudeの方は5時間の方がデータが取れていないと出てしまうのですが、何か追加の設定が必要でしょうか。ときた。バグっぽいけど原因調べて。

> はい、全部直して！！あと、なんで私のところでは動いているの？

（後者への回答: プローブが成功する環境では Claude に窓が入るので注記は出ず、`marked` が true になって両方に記号が描かれる。1 の誤読は「Claude の窓が 0 本 かつ Codex に窓がある」ときにしか表に出ない。2 の固着もプローブが完走する環境では起きない。）

## 実装

- `src/composables/rateLimitGauge.ts` — `rateLimitReadout(snapshot, now_ms) → { note, gauges }` を追加。`agentGauges` / `claudeProbeNote` は export をやめて内部に閉じた。呼び出し側が注記と記号を別々に決められる限り同じズレが再発するため、1 つの関数が両方を返す形にしてある。
- `src/components/RateLimitGauge.vue` — 既存の 1 パス `view` computed をそのまま差し替え。
- `server/agents/statusline.ts` — `hadApiResponse()` と `readClaudeStatus()` を追加。
- `server/agents/rate-limit-store.ts` — `reportCodex` / `reportClaudeStatus` に分割。API 応答前の窓なし statusLine は状態も `lastStatusLineAt_ms` も動かさないので、プローブが黙って終われば正しく `no-report` の指数バックオフに乗る。
- `docs/guide/{en,ja}/features.md` — 行に 2 つ以上並ぶときは数字ごとに記号が付くことを追記（日英同期）。
- `plans/fix-1161-rate-limit-readout.md` — 調査と設計。

## テスト

- 追加した 2 本（composable と描画）が、`marked` を元の式に戻すと**両方落ちる**ことを確認済み。
- `claudeStatusVerdict` の 3 ケース、`hadApiResponse` / `readClaudeStatus`（実測した 2 つのペイロード形をそのまま使用）、API 応答前 statusLine → タイムアウトが `no-report` failures:1 になること、その後 windows が届けば成功すること。
- `yarn format` / `lint` / `typecheck` / `typecheck:server` / `typecheck:test` / `build` / `test`（457 files, 6731 tests）すべて green。

work in mulmoterminal

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved rate-limit reporting to clearly distinguish Claude and Codex usage.
  * Prevented preliminary Claude status messages from incorrectly ending rate-limit checks.
  * Preserved valid usage readings when one tool has unavailable or incomplete data.
  * Updated labels and indicators so displayed percentages identify the associated tool.

* **Documentation**
  * Clarified rate-limit gauge behavior in English and Japanese documentation.

### CODE REVIEW

File: docs/guide/en/features.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ MulmoTerminal — a browser terminal for parallel Claude Code and Codex agents 
...
-| Rate-limit gauge | The **5h / 7d** windows your Claude (and Codex) subscription shares across every session, always visible in the grid header. Running many agents is what burns them fastest, and nothing else in the app showed them. Shown only once an agent reports; hover for when each window resets. **When it cannot be shown, it says why** — no `claude` on PATH, an account with no windows (API-key billing), or a check that got no answer and is retrying |
+| Rate-limit gauge | The **5h / 7d** windows your Claude (and Codex) subscription shares across every session, always visible in the grid header. Running many agents is what burns them fastest, and nothing else in the app showed them. Shown only once an agent reports; hover for when each window resets. Whenever more than one thing is on the row, each set of figures carries its tool's mark, so `claude usage n/a` beside a `7d` figure cannot be mistaken for Claude's own. **When it cannot be shown, it says why** — no `claude` on PATH, an account with no windows (API-key billing), or a check that got no answer and is retrying |
```
[the remark this answers is not in the document]
Comment (isamu) [reply docs/guide/en/features.md:27]: Skipping in this PR, with reasoning — not disagreeing that the change is user-visible.

The screenshot rule in `CLAUDE.md` sits under **Publishing a release → "2. A dated setup guide, `docs/guide/{en,ja}/v<version>.md`"**, and its own naming instruction is `v<version>-<thing>.png`. This PR adds no release page and the fix is unreleased, so there is no version to name the file after — following the instruction literally is not possible yet. Guessing a number is exactly how a filename rots.

Two more reasons it belongs on the release page rather than here:

- The state being illustrated is a **failed** Claude probe (that is the only condition under which the note and a Codex figure share the row). A failure state is the right thing to show in a dated "what changed and how to tell you have it" page; it is the wrong canonical image for a reference-table row that describes the gauge in general.
- Capturing it needs a real server with a scratch `HOME` (the repo rule about screenshots leaking maintainer paths), plus a deliberately broken `CLAUDE_BIN`. That setup is release-guide work.

Committing to it: whichever version ships this gets `docs/guide/{en,ja}/v<version>.md` with `docs/guide/images/v<version>-rate-limit-readout.png`, captured against a real server at `deviceScaleFactor: 2` and downscaled, per the release checklist.

The prose clause stays because it answers the question the reporter actually had — whether the figure next to `claude usage n/a` is Claude's.

**Comment by isamu** [comment]
## Bot review triage — iteration 1

| Reviewer | Signal |
| --- | --- |
| Codex (`CODEX VERDICT`) | **LGTM** — no correctness / security / a11y / edge-case / test-coverage findings |
| CodeRabbit | 1 actionable (Minor): missing release screenshot |
| Socket Security | Project Report + PR Alerts pass |
| CI | all checks green (lint-and-build ubuntu/macos, package-smoke, knip, jscpd, triage) |

### Addressed vs. skipped

- **Skipped with reasoning** — CodeRabbit's screenshot request ([reply](https://github.com/receptron/mulmoterminal/pull/1162#discussion_r3686886674)). The rule is scoped to dated release guide pages and names files `v<version>-<thing>.png`; this fix is unreleased, so there is no version to name one after. Committed to capturing it on the release page that ships this.

No code changes this iteration.

### Observed during my own review, not flagged by either bot

**The retry cadence changes for a probe that boots and then never finishes.** Measured against the store:

| | first attempt → next attempts |
| --- | --- |
| before | flat **60 min** from the first failure (misclassified as `no-windows`) |
| after | 90s → 3m → 6m → 12m → 24m → 48m → 60 min ceiling |

That is the `no-report` backoff #1011 designed, and it now applies because the failure is finally classified as one. It cuts both ways and both are improvements:

- A **transient** failure (machine busy, TUI slow to boot) used to leave the gauge dead for an hour; it now recovers in 90 seconds.
- A **persistent** one costs about 6 probes over the first ~1.5 hours instead of 2, then settles at the same ceiling as before.

Probes only run while a browser is polling the gauge, so an idle machine still spends nothing. Flagging it because it is a real change in what the feature costs, and it is not visible from the diff alone.

**If `cost.total_api_duration_ms` ever disappears upstream**, `hadApiResponse` answers `false` forever. The consequence is bounded: the success path is unaffected (windows in hand short-circuits), and an API-key account falls into the `no-report` backoff whose ceiling is the same 60 min `no-windows` used — only the hover wording degrades from "API-key billing" to "Retrying". That is the direction the default was chosen to fail in.
